src/healthData/HealthDataManager.js

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2015 Adobe Systems Incorporated. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, maxerr: 50 */
+/*global define, $, brackets, console */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var AppInit = require("utils/AppInit"),
+        PreferencesManager = require("preferences/PreferencesManager"),
+        HealthDataUtils = require("healthData/HealthDataUtils"),
+        uuid = require("thirdparty/uuid"),
+        UrlParams = require("utils/UrlParams").UrlParams;
+
+    var prefs = PreferencesManager.getExtensionPrefs("healthData");
+
+    prefs.definePreference("healthDataTracking", "boolean", true);
+
+    var ONE_DAY = 24 * 60 * 60 * 1000,
+        timeoutVar;
+
+    var params = new UrlParams();
+    params.parse();
+
+    /**
+     * Get the health data which will be send to the server. Initially it is only one time data.
+     */
+    function getHealthData() {
```

Comment:
As this gets more complex, I think it would be good to split this into a different file: one module manages deciding when to send, and doing the actual network IO, and a separate module manages collecting the data & rolling it up into a JSON blob.  E.g. maybe HealthDataCollector & HealthDataUploader?